deepchem/models/torch_models/neuralode.py

```diff
@@ -0,0 +1,218 @@
+        ode_block=None,
+        enable_ffn=False)
+
+    self.neural_ode = NeuralODE(vector_field=self.vector_field,
+                                solver=solver,
+                                order=order,
+                                atol=atol,
+                                rtol=rtol,
+                                sensitivity=sensitivity,
+                                solver_adjoint=solver_adjoint,
+                                atol_adjoint=atol_adjoint,
+                                rtol_adjoint=rtol_adjoint,
+                                interpolator=interpolator,
+                                integral_loss=integral_loss,
+                                seminorm=seminorm,
+                                return_t_eval=return_t_eval,
+                                optimizable_params=optimizable_params)
+
+    self.model = LinearModule(n_features=n_features,
+                              layer_filters=layer_filters,
+                              dropouts=dropouts,
+                              activation_fns=activation_fns,
+                              weight_init_stddevs=weight_init_stddevs,
+                              bias_init_consts=bias_init_consts,
+                              residual=residual,
+                              ode_block=self.neural_ode,
+                              enable_ffn=True)
+
+    self.t_span = t_span
+
+    regularization_loss: Optional[Callable]
+
+    if weight_decay_penalty != 0:
+      weights = [layer.weight for layer in self.model.layers]
+      if weight_decay_penalty_type == 'l1':
+        regularization_loss = lambda: weight_decay_penalty * torch.sum(
+            torch.stack([torch.abs(w).sum() for w in weights]))
+      else:
+        regularization_loss = lambda: weight_decay_penalty * torch.sum(
+            torch.stack([torch.square(w).sum() for w in weights]))
+    else:
+      regularization_loss = None
+
+    loss = L2Loss()
+
+    output_types = ['prediction']
+
+    super(NeuralODEModel,
+          self).__init__(self.model,
+                         loss=loss,
+                         output_types=output_types,
+                         regularization_loss=regularization_loss,
+                         **kwargs)
+
+  def default_generator(self,
+                        dataset,
+                        epochs=1,
+                        mode='fit',
+                        deterministic=True,
+                        pad_batches=True):
+
+    for epoch in tqdm(range(epochs)):
+      for (t_b, Y_b, w_b,
+           ids_b) in dataset.iterbatches(batch_size=self.batch_size,
+                                         deterministic=deterministic,
+                                         pad_batches=pad_batches):
+        if mode == 'predict':
+          dropout = np.array(0.0)
+        else:
+          dropout = np.array(1.0)
+        yield ([t_b, dropout], Y_b, [w_b])
+
+  # TODO : Write custom fit method
```

Comment:
We probably shouldn't have a custom fit method. Try double checking that the regular fit won't work